packages/components/src/mobile/global-styles-context/utils.native.js

```diff
@@ -226,14 +232,31 @@ export function parseStylesVariables( styles, mappedValues, customValues ) {
 				] );
 			} );
 		}
+
+		if ( variable === 'var' ) {
+			stylesBase = stylesBase.replace( varRegex, ( _$1, $2 ) => {
+				if ( mappedValues && mappedValues?.color ) {
+					const matchedValue = find( mappedValues.color?.values, {
+						slug: $2,
+					} );
+					return `"${ matchedValue?.color }"`;
+				}
+				return UNKNOWN_VALUE;
+			} );
+		}
 	} );
 
 	return JSON.parse( stylesBase );
 }
 
 export function getMappedValues( features, palette ) {
 	const typography = features?.typography;
-	const colors = { ...palette?.theme, ...palette?.custom };
+	const colors = [
+		...( palette?.theme ? palette.theme : [] ),
+		...( palette?.custom ? palette.custom : [] ),
+		...( palette?.default ? palette.default : [] ),
```

Comment:
I think we could reduce these lines with:
```suggestion
		...( palette?.theme || [] ),
		...( palette?.custom || [] ),
		...( palette?.default || [] ),
```

Reply:
Thanks for the suggestion! I applied the change 👍